tmt/result.py

```diff
@@ -189,6 +189,14 @@ class CheckResult(BaseResult):
         serialize=lambda event: event.value,
         unserialize=CheckEvent.from_spec)
 
+    def to_subcheck(self) -> 'SubCheckResult':
+        """ Convert check to a tmt SubCheckResult """
+
+        # TODO: Try to find a better way for conversion?
+        # TODO: Some attributes will be probably modified for SubCheckResult e.g. name (namespace)
+        # TODO: Still need to check if this actually works
```

Comment:
So, does it work? :)

Reply:
Yes, it seems to work as expected:

```
ipdb> results[0]
CheckResult(name='dmesg', result=<ResultOutcome.SKIP: 'skip'>, original_result=<ResultOutcome.SKIP: 'skip'>, note=None, log=[], start_time='2024-10-15T20:53:46.202625+00:00', end_time='2024-10-15T20:53:46.202653+00:00', duration='00:00:00', event=<CheckEvent.BEFORE_TEST: 'before-test'>)
ipdb> results[0].to_subcheck()
SubCheckResult(name='dmesg', result=<ResultOutcome.SKIP: 'skip'>, original_result=<ResultOutcome.SKIP: 'skip'>, note=None, log=[], start_time='2024-10-15T20:53:46.202625+00:00', end_time='2024-10-15T20:53:46.202653+00:00', duration='00:00:00', event=<CheckEvent.BEFORE_TEST: 'before-test'>)
```

I tried to convert `Result` to `SubResult` the same way, but it's necessary to filter in only some of the `Result` fields. Would you mind to take a look?
- https://github.com/teemtee/tmt/pull/3200/commits/978fe968cb26d976061a803e8d54e0ad5376b448

Or if you think the direct mapping of the Result fields to SubResult ones is better, just let me know and I'll revert the commit.

Thanks :)

Reply:
- [ ] TODO: Try to get the SubResult fields from the class in `to_subresult` method, do not define them explicitly. Try to use "container_*" methods from `tmt.utils`.